crates/forge/src/init.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::{bail, Context, Ok, Result};
+
+use include_dir::{include_dir, Dir};
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use toml_edit::{ArrayOfTables, Document, Item, Table};
+
+static TEMPLATE: Dir = include_dir!("starknet_forge_template");
+
+fn overwrite_files_from_template(
+    dir_to_overwrite: &str,
+    base_path: &PathBuf,
+    project_name: &str,
+) -> Result<()> {
+    let copy_form_dir = TEMPLATE.get_dir(dir_to_overwrite);
+    match copy_form_dir {
+        Some(dir) => {
+            for file in dir.files() {
+                fs::create_dir_all(base_path.join(Path::new(dir_to_overwrite)))?;
+                let path = base_path.join(file.path());
+                let contents = file.contents();
+                let contents = replace_project_name(contents, project_name);
+
+                fs::write(path, contents)?;
+            }
+        }
+        None => {
+            bail!(
+                "Directory {} doesn't exist in the template.",
+                dir_to_overwrite
+            )
+        }
+    }
+
+    Ok(())
+}
+
+fn replace_project_name(contents: &[u8], project_name: &str) -> Vec<u8> {
+    // SAFETY: We control these files, and we know that they are UTF-8.
+    let contents = unsafe { std::str::from_utf8_unchecked(contents) };
+    let contents = contents.replace("{{ PROJECT_NAME }}", project_name);
+    contents.into_bytes()
+}
+
+fn add_target_to_toml(path: &PathBuf) -> Result<()> {
+    let config_file = fs::read_to_string(path)?;
+
+    let mut doc = config_file.parse::<Document>().expect("invalid document");
+    let mut array_of_tables = ArrayOfTables::new();
+    let mut casm = Table::new();
+    let mut contract = Table::new();
+    contract.set_implicit(true);
+
+    casm.insert("casm", Item::Value(true.into()));
+    array_of_tables.push(casm);
+    contract.insert("starknet-contract", Item::ArrayOfTables(array_of_tables));
+    doc.insert("target", Item::Table(contract));
+
+    fs::write(path, doc.to_string())?;
+
+    Ok(())
+}
+
+pub fn init(name: Option<String>) -> Result<()> {
+    let project_name = name.unwrap_or("starknet_forge_template".to_string());
+    let project_path = std::env::current_dir()?.join(&project_name);
+
+    Command::new("scarb")
+        .current_dir(std::env::current_dir().context("Failed to get current directory")?)
+        .arg("new")
+        .arg(&project_path)
+        .stderr(Stdio::inherit())
+        .stdout(Stdio::inherit())
+        .output()
+        .context("Failed to initial new project")?;
+
+    Command::new("scarb")
+        .current_dir(&project_path)
+        .arg("add")
+        .arg("snforge_std")
+        .arg("--git")
+        .arg("https://github.com/foundry-rs/starknet-foundry.git")
```

Comment:
The problem here is that we add `snforge_std` version from main branch - we should add it with a tag pointing to the version of forge from Cargo.toml. 

The approach for tests should be different though - we should overwrite this dependency in tests to point to the current branch and repository (we should duplicate the logic from git dependency test, check updates made by @cptartur)